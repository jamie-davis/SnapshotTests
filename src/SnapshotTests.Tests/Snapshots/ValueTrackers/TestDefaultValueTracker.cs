using System;
using System.Linq;
using NUnit.Framework;
using SnapshotTests.Snapshots.ValueTrackers;
using TestConsoleLib;
using TestConsoleLib.Testing;

namespace SnapshotTests.Tests.Snapshots.ValueTrackers
{
    [TestFixture]
    public class TestDefaultValueTracker
    {
        [Test]
        public void ValuesAreSubstituted()
        {
            //Arrange
            var values = new object[]
            {
                "Value 1",
                2,
                DateTime.Parse("2020-07-12 10:38"),
                "Value 1",
                2,
                DateTime.Parse("2020-07-12 10:38"),
                "Value 2",
                3,
                DateTime.Parse("2020-07-12 10:39"),
                "Value 2",
                3,
                DateTime.Parse("2020-07-12 10:39"),
                Guid.Parse("F1684874-FF8C-4D0D-A767-E1EC90F36F8D"),
                null,
                null
            };

            var sub = new DefaultValueTracker("TestCol");

            //Act
            var result = values.Select(v => new { Value = v, Substitute = sub.GetSubstitute(v)});

            //Assert
            var output = new Output();

            output.WrapLine("Substitutions:");
            output.WriteLine();
            output.FormatTable(result);

            output.WriteLine();
            output.WrapLine("Unique Substitution Values:");
            output.WriteLine();
            output.FormatTable(result.Select(r => r.Substitute).Distinct().Select(s => new { UniqueValue = s }));

            output.Report.Verify();
        }
    }
}
