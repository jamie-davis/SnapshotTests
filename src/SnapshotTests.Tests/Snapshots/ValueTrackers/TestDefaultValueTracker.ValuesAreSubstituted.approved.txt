Substitutions:                                                                                 
                                                                                               
Value                                Substitute                                                
------------------------------------ ----------                                                
Value 1                              TestCol-1                                                 
2                                    TestCol-2                                                 
12/07/2020 10:38:00                  TestCol-3                                                 
Value 1                              TestCol-1                                                 
2                                    TestCol-2                                                 
12/07/2020 10:38:00                  TestCol-3                                                 
Value 2                              TestCol-4                                                 
3                                    TestCol-5                                                 
12/07/2020 10:39:00                  TestCol-6                                                 
Value 2                              TestCol-4                                                 
3                                    TestCol-5                                                 
12/07/2020 10:39:00                  TestCol-6                                                 
f1684874-ff8c-4d0d-a767-e1ec90f36f8d TestCol-7                                                 
                                                                                               
                                                                                               
                                                                                               
Unique Substitution Values:                                                                    
                                                                                               
Unique                                                                                         
Value                                                                                          
---------                                                                                      
TestCol-1                                                                                      
TestCol-2                                                                                      
TestCol-3                                                                                      
TestCol-4                                                                                      
TestCol-5                                                                                      
TestCol-6                                                                                      
TestCol-7                                                                                      
                                                                                               
                                                                                               